app/controllers/carts_controller.rb

```diff
@@ -1,15 +1,20 @@
 class CartsController < ApplicationController
+  before_action :logged_in_user, only: :add_to_cart
 
   def index
     @cart = get_cart
   end
 
   def add_to_cart
-    if logged_in?
-      add_product_to_cart_cookie(params[:product_id])
-      redirect_to root_path
-    else
-      redirect_to login_path
-    end
+    add_product_to_cart_cookie(params[:product_id])
   end
+
+  private 
+    
+    def logged_in_user
```

Comment:
this method name looks like a getter, and instead it'd doing an action - I think it should be renamed to show that it's an action, like `check_logged_in_user`. Does it make sense?

Reply:
Yes, the method name is from the book, I think it remained like this from when the before_action method was named before_filter and that made more sense, to have a before_filter logged_in_user. I will rename it.

Reply:
👍🏻